app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderPresenter.kt

```diff
@@ -606,6 +607,15 @@ class ReaderPresenter(
         return viewerChaptersRelay.value?.currChapter
     }
 
+    fun getSource() = manga?.source?.let { sourceManager.getOrStub(it) } as? HttpSource
+
+    fun getChapterUrl(): String? {
+        val sChapter = getCurrentChapter()?.chapter?.toDomainChapter()?.toSChapter() ?: return null
```

Comment:
```suggestion
        val sChapter = getCurrentChapter()?.chapter ?: return null
```
DbChapter inherits SChapter